### ISSUE TITLE
If VS Code intellisense supports clang-cl.exe then use that instead of cl.exe

### ISSUE THREAD

**Issue by ghost** [KeyWorksRW/ttBld#108]
### Description:
The main problem with using cl.exe for intellisense is that the path to the compiler changes every single time the user updates Visual Studio. According to the intellisense tooltip, c_cpp_properties.json is supposed to contain the full path to the compiler. While **MakeNinja** can automatically locate the new path and update the property file, it does require the user to always run **MakeNinja** before launching VS Code. We do provide a batch file that does exactly that, but we can't assume the user will always do that.

If intellisense works with the clang-cl.exe compiler, and it's available on the user's system, we should use that instead. If that's not available, we could also check to see if the GCC compiler is available (assuming it works with intellisense). Both of these compilers only change their path when there is a major update — and even then the user may simply install the udpate in the same location so that no PATH changes are required.

Bottom line is that if VS Code Intellisense works on a non-MSVC compiler that the user has installed, then we should setup the property sheet to use that so that Intellisense will always just work no matter how many times MS changes their compiler's path.

### Issue Type:
- [ ] Bug report
- [x] Feature request
- [ ] Design change request
- [ ] Other
